### PR TITLE
HBASE-26691 Replacing log4j with reload4j for branch-2.x

### PR DESCRIPTION
**Pull request by Apache9**

### CODE REVIEW

**Review by jojochuang — COMMENTED** [review]
does it also ban transitive dependencies on log4j and slf4j-log4j12?

**Comment by Apache9** [comment]
> does it also ban transitive dependencies on log4j and slf4j-log4j12?

Yes, if you do not want to ban transitive dependency, just add `<searchTransitive>false</searchTransitive>`

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  4s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   9m 22s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 40s |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 25s |  the patch passed  |
| +1 :green_heart: |  javac  |   9m 25s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 49s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  12m 49s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   7m  7s |  The patch does not generate ASF License warnings.  |
|  |   |  61m 27s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | dupname asflicense javac hadoopcheck xml compile |
| uname | Linux 77ad41f5b7e1 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / d9a20630b0 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 141 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 31s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 35s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   3m  1s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   7m 17s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  14m 27s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  4s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  4s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 37s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  14m 27s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 174m  7s |  root in the patch failed.  |
|  |   | 242m 42s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 982f6a27e547 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / d9a20630b0 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/testReport/ |
| Max. process+thread count | 5878 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/1/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 46s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 29s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   5m  3s |  branch-2 passed  |
| +1 :green_heart: |  compile  |  12m 29s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   5m  4s |  the patch passed  |
| +1 :green_heart: |  compile  |  11m 58s |  the patch passed  |
| +1 :green_heart: |  javac  |  11m 58s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 54s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  16m 55s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   8m  6s |  The patch does not generate ASF License warnings.  |
|  |   |  77m 54s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | dupname asflicense javac hadoopcheck xml compile |
| uname | Linux b8ecb014db5c 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / d9a20630b0 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 141 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 13s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m  7s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 42s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   7m 18s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  11m 23s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 59s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 45s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 45s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m  8s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  11m 49s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  | 344m 48s |  root in the patch failed.  |
|  |   | 405m 24s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 46f1969ace8f 4.15.0-153-generic #160-Ubuntu SMP Thu Jul 29 06:54:29 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / d9a20630b0 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/testReport/ |
| Max. process+thread count | 2346 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/2/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by jojochuang** [comment]
The latest reload4j version is 1.2.18.3 with a few additional vulnerabilities fixed.

**Comment by Apache-HBase** [comment]
:confetti_ball: **+1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 37s |  Docker mode activated.  |
||| _ Prechecks _ |
| +1 :green_heart: |  dupname  |   0m  1s |  No case conflicting files found.  |
| +1 :green_heart: |  @author  |   0m  0s |  The patch does not contain any @author tags.  |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 25s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   3m 50s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   9m 27s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   3m 37s |  the patch passed  |
| +1 :green_heart: |  compile  |   9m 30s |  the patch passed  |
| +1 :green_heart: |  javac  |   9m 30s |  the patch passed  |
| +1 :green_heart: |  whitespace  |   0m  0s |  The patch has no whitespace issues.  |
| +1 :green_heart: |  xml  |   0m 49s |  The patch has no ill-formed XML file.  |
| +1 :green_heart: |  hadoopcheck  |  12m 48s |  Patch does not cause any errors with Hadoop 3.1.2 3.2.1.  |
||| _ Other Tests _ |
| +1 :green_heart: |  asflicense  |   8m  6s |  The patch does not generate ASF License warnings.  |
|  |   |  62m  1s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/artifact/yetus-general-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | dupname asflicense javac hadoopcheck xml compile |
| uname | Linux 090e7ff20101 4.15.0-112-generic #113-Ubuntu SMP Thu Jul 9 23:41:39 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 3021c58513 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| Max. process+thread count | 141 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   0m 35s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  6s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 26s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 29s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 59s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   7m 15s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  14m 20s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 22s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m 14s |  the patch passed  |
| +1 :green_heart: |  compile  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  javac  |   3m  1s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   7m 39s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  14m 23s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  15m  5s |  root in the patch failed.  |
|  |   |  81m 29s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/artifact/yetus-jdk11-hadoop3-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux 9673f80ff259 4.15.0-58-generic #64-Ubuntu SMP Tue Aug 6 11:12:41 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 3021c58513 |
| Default Java | AdoptOpenJDK-11.0.10+9 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/artifact/yetus-jdk11-hadoop3-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/testReport/ |
| Max. process+thread count | 836 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

**Comment by Apache-HBase** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| +0 :ok: |  reexec  |   1m 12s |  Docker mode activated.  |
| -0 :warning: |  yetus  |   0m  7s |  Unprocessed flag(s): --brief-report-file --spotbugs-strict-precheck --whitespace-eol-ignore-list --whitespace-tabs-ignore-list --quick-hadoopcheck  |
||| _ Prechecks _ |
||| _ branch-2 Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 23s |  Maven dependency ordering for branch  |
| +1 :green_heart: |  mvninstall  |   4m 10s |  branch-2 passed  |
| +1 :green_heart: |  compile  |   2m 40s |  branch-2 passed  |
| +1 :green_heart: |  shadedjars  |   7m 11s |  branch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  11m 31s |  branch-2 passed  |
||| _ Patch Compile Tests _ |
| +0 :ok: |  mvndep  |   0m 20s |  Maven dependency ordering for patch  |
| +1 :green_heart: |  mvninstall  |   4m  0s |  the patch passed  |
| +1 :green_heart: |  compile  |   2m 43s |  the patch passed  |
| +1 :green_heart: |  javac  |   2m 43s |  the patch passed  |
| +1 :green_heart: |  shadedjars  |   8m 10s |  patch has no errors when building our shaded downstream artifacts.  |
| +1 :green_heart: |  javadoc  |  11m 43s |  the patch passed  |
||| _ Other Tests _ |
| -1 :x: |  unit  |  27m 16s |  root in the patch failed.  |
|  |   |  87m  0s |   |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | ClientAPI=1.41 ServerAPI=1.41 base: https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/artifact/yetus-jdk8-hadoop2-check/output/Dockerfile |
| GITHUB PR | https://github.com/apache/hbase/pull/4050 |
| Optional Tests | javac javadoc unit shadedjars compile |
| uname | Linux eeb4f71f6350 4.15.0-162-generic #170-Ubuntu SMP Mon Oct 18 11:38:05 UTC 2021 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | dev-support/hbase-personality.sh |
| git revision | branch-2 / 3021c58513 |
| Default Java | AdoptOpenJDK-1.8.0_282-b08 |
| unit | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/artifact/yetus-jdk8-hadoop2-check/output/patch-unit-root.txt |
|  Test Results | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/testReport/ |
| Max. process+thread count | 716 (vs. ulimit of 12500) |
| modules | C: hbase-logging hbase-common hbase-metrics-api hbase-hadoop-compat hbase-metrics hbase-hadoop2-compat hbase-client hbase-zookeeper hbase-replication hbase-http hbase-asyncfs hbase-procedure hbase-server hbase-mapreduce hbase-testing-util hbase-thrift hbase-rsgroup hbase-shell hbase-endpoint hbase-it hbase-rest hbase-examples hbase-shaded hbase-shaded/hbase-shaded-client hbase-hbtop hbase-compression/hbase-compression-aircompressor hbase-compression/hbase-compression-lz4 hbase-compression/hbase-compression-snappy hbase-compression/hbase-compression-xz hbase-compression/hbase-compression-zstd hbase-assembly hbase-shaded/hbase-shaded-testing-util hbase-shaded/hbase-shaded-testing-util-tester hbase-shaded/hbase-shaded-check-invariants hbase-shaded/hbase-shaded-with-hadoop-check-invariants hbase-archetypes/hbase-client-project hbase-archetypes/hbase-shaded-client-project . U: . |
| Console output | https://ci-hadoop.apache.org/job/HBase/job/HBase-PreCommit-GitHub-PR/job/PR-4050/3/console |
| versions | git=2.17.1 maven=3.6.3 |
| Powered by | Apache Yetus 0.12.0 https://yetus.apache.org |


This message was automatically generated.

File: hbase-mapreduce/pom.xml [hunk 1]
```diff
@@ -426,42 +350,6 @@
       </dependencies>
 
     </profile>
-
```
Comment (jojochuang) [inline hbase-mapreduce/pom.xml:429]: I'm not sure about this part, but the rest LGTM

**Comment by jojochuang** [comment]
Basically in addition to replacing reload4j, the PR also simplifies dependency management.

**Comment by jojochuang** [comment]
can you also verify the test failure is unrelated?

**Comment by jojochuang** [comment]
I am testing my Hadoop reload4j patch, and then realized there are a lot of gotchas when building downstream applications.
Please try build HBase downstream applications with this change. I tried to merge this PR in my internal branch and then test but we're too far away from branch-2 and resulted lots of conflicts.

**Comment by Apache9** [comment]
> I am testing my Hadoop reload4j patch, and then realized there are a lot of gotchas when building downstream applications. Please try build HBase downstream applications with this change. I tried to merge this PR in my internal branch and then test but we're too far away from branch-2 and resulted lots of conflicts.

For HBase we have tried out best to not introduce log4j dependencies to our downstream users, they will only pull in slf4j, so I do not think it will break our users too much. But it is true that hadoop and zookeeper will introduce log4j and slf4j-log4j12 dependencies to downstreamer users, I suggest we do a round of cleanup, to avoid introducing these dependencies.

Thanks.

File: hbase-mapreduce/pom.xml [hunk 2]
```diff
@@ -220,6 +220,24 @@
       <scope>compile</scope>
       <optional>true</optional>
     </dependency>
+    <dependency>
```
Comment (apurtell) [inline hbase-mapreduce/pom.xml:223]: Is this related?

File: hbase-mapreduce/pom.xml [hunk 3]
```diff
@@ -322,100 +340,6 @@
         </property>
       </activation>
       <dependencies>
-        <dependency>
```
Comment (apurtell) [inline hbase-mapreduce/pom.xml:325]: Oh, I see, some extra refactoring?

File: hbase-shaded/hbase-shaded-testing-util/pom.xml [hunk 4]
```diff
@@ -37,43 +37,8 @@
         <dependency>
             <groupId>org.apache.hadoop</groupId>
             <artifactId>hadoop-common</artifactId>
-            <version>${hadoop.version}</version>
             <type>test-jar</type>
             <scope>compile</scope>
-            <exclusions>
```
Comment (apurtell) [inline hbase-shaded/hbase-shaded-testing-util/pom.xml:43]: I don't have a strong opinion about this so will approve the PR but it would be better to stick with just the changes that are implied by the description _[HBASE-26691 Replacing log4j with reload4j for branch-2.x]_

**Review by apurtell — APPROVED** [review]

**Comment by apurtell** [comment]
If there are no more concerns or updates, I will merge this tomorrow.

**Comment by Apache9** [comment]
So do we plan to use reload4j for 2.x, or we plan to directly upgrade to log4j2?
I plan to backport the log4j2 changes to branch-2 and try to see if using the old log4j.properties is fine with log4j-1.2-api in place.

**Comment by busbey** [comment]
how about we get this in place for branches-2 now, so that we can stop shipping hbase 2 releases with an EOL log4j 1, but we plan for the log4j 2 changes to be the preferred path for whatever the next minor 2.x release is after the backport lands. (presuming the testing goes well.)

**Comment by Apache9** [comment]
> how about we get this in place for branches-2 now, so that we can stop shipping hbase 2 releases with an EOL log4j 1, but we plan for the log4j 2 changes to be the preferred path for whatever the next minor 2.x release is after the backport lands. (presuming the testing goes well.)

Fine by me~

**Comment by apurtell** [comment]
Merged. Agreed, we can do this now and I will pick it all the way back to branch-2.4, and anytime we can commit log4j2 work when ready to branch-2.5 and branch-2.

**Comment by apurtell** [comment]
I also committed an addendum that ups the reload4j version to the latest release at this time, 1.2.19, and fixes a couple of misses in shading and assembly configuration.

File: hbase-mapreduce/pom.xml [hunk 5]
```diff
@@ -220,6 +220,24 @@
...
+    <dependency>
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-mapreduce-client-core</artifactId>
+    </dependency>
+    <dependency>
+      <!--maven dependency:analyze says not needed but tests fail w/o-->
+      <groupId>org.apache.hadoop</groupId>
+      <artifactId>hadoop-minicluster</artifactId>
+      <scope>test</scope>
+    </dependency>
```
Comment (lizhexi) [inline hbase-mapreduce/pom.xml:240]: The child POM does not inherit the exclusion of the dependency introduced by the parent POM. Is the reconstruction safe? What are the considerations for deleting an exclusion?
